### PR TITLE
Add Curriculumbuilder resourcelink

### PR DESCRIPTION
**Pull request by Hamms**
As defined in https://github.com/mrjoshida/curriculumbuilder/blob/bf74aa5/curriculumBuilder/resourcelinks.py

### CODE REVIEW

File: src/plugins/resourcelink.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+let redact;
+
+const RESOURCELINK_RE = /^\[r ([^\]]+)\]/;
+const RESOURCELINK = 'resourcelink';
+
+/**
+ * Plugin that adds support for Curriculum Builder's resourcelinks.
```
Comment (Hamms) [inline src/plugins/resourcelink.js]: See https://github.com/remarkjs/remark/tree/master/packages/remark-parse#extending-the-parser for more on creating remark plugins

**Comment by davidsbailey** [comment]
👀 looking...

File: test/unit/plugins/resourcelink.test.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  describe('restore', () => {
+    it('can restore resourcelinks back to markdown', () => {
+      const source = "[r some-slug]";
+      const redacted = "[][0]"
+      const output = parser.sourceAndRedactedToMarkdown(source, redacted);
+      expect(output).toEqual("[r some-slug]\n");
+    });
+
+    it('cannot resotre resourcelinks to html', () => {
```
Comment (davidsbailey) [inline test/unit/plugins/resourcelink.test.js]: nit: typo in `restore`

File: test/unit/plugins/resourcelink.test.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+const expect = require('expect');
+const parser = require('../../../src/cdoFlavoredParser');
+
+describe('resourcelink', () => {
+  describe('render', () => {
+    it('cannot render resourcelinks to html', () => {
+      const input = "[r some-slug]";
+      const output = parser.sourceToHtml(input);
+      expect(output).toEqual("<p>[r some-slug]</p>\n");
+    });
```
Comment (davidsbailey) [inline test/unit/plugins/resourcelink.test.js]: why does this test case prove that you cannot render to html -- what would this output look like if you could?

**Comment by Hamms** [comment]
Oh, I probably should have offered to do an in-person before asking you to review! Lemme know anytime you'd like and overview and I'd be happy to do a braindump

**Pull request merged**